src/groups/mqb/mqbblp/mqbblp_cluster.cpp

```diff
@@ -3694,21 +3691,41 @@ void Cluster::gcQueueOnDomainDispatched(mqbcmd::ClusterResult* result,
     BSLS_ASSERT_SAFE(dispatcher()->inDispatcherThread(this));
 
     // 'true' implies immediate
-    if (const int rc = d_clusterOrchestrator.queueHelper().gcExpiredQueues(
-            true,
-            domainName)) {
+    const int rc =
+        d_clusterOrchestrator.queueHelper().gcExpiredQueues(true, domainName);
+    if (rc == -1 || rc == -3) {
+        // TBD: We allow the node to not be an active primary for *any*
```

Comment:
Good note.  Leader != primary will break a lot of assumptions, so it's good to document them as we add them.